Update weld-se-core to 3.1.1.Final

Hi

Currently no one is able to use weld-junit5 2.0.0.Final as it is broken by weld-se-core 3.1.0.Final.

The new version of weld-se-core 3.1.1.Final fixes this issue and will appreciate if someone could update weld-se-core to 3.1.1.Final.

The issue I am seeing is;
java.lang.nosuchmethoderror: javax.enterprise.inject.spi.observermethod.getpriority()i

BR
Lulseged Zerfu

Hi

I am not sure if it helps to upgrade!
I could not find it. Let me show what I am doing:

I am using java 12.0.1
Junit 5.4.2
weld-junit 2.0.0.Final

// main class
public class LoggerProducer implements Serializable
{
    @Produces
    public Logger getLogger(final InjectionPoint injectionPoint)
    {
        return LogManager.getLogger(injectionPoint.getMember().getDeclaringClass().getName());
    }
}

// unit testing it
@ExtendWith(WeldJunit5AutoExtension.class)
class LoggerProducerTest
{
    @Inject
    private Logger logger;

    @WeldSetup
    private static WeldInitiator weldInitiator = WeldInitiator.of(LoggerProducer.class);

    @BeforeAll
    static void beforeAll()
    {

    }

    @AfterAll
    static void afterAll()
    {

    }

    @BeforeEach
    void setup()
    {
        logger.trace("before each");
    }

    @AfterEach
    void tearDown()
    {
        logger.trace("after each");
    }

    @Test
    void testLogger()
    {
        assertNotNull(logger);
    }
}

The following is what I am seeing in the console:
[INFO] Running biz.shb.inventory.test.core.logger.LoggerProducerTest
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.jboss.weld.util.bytecode.ClassFileUtils$1 (file:/d:/repository/maven/org/jboss/weld/weld-core-impl/3.1.0.Final/weld-core-impl-3.1.0.Final.jar) to method java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int)
WARNING: Please consider reporting this to the maintainers of org.jboss.weld.util.bytecode.ClassFileUtils$1
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.601 s <<< FAILURE! - in biz.shb.inventory.test.core.logger.LoggerProducerTest
[ERROR] testLogger  Time elapsed: 0.586 s  <<< ERROR!
org.jboss.weld.exceptions.WeldException: WELD-001524: Unable to load proxy class for bean Implicit Bean [javax.enterprise.inject.Instance] with qualifiers [@Default] with class interface javax.enterprise.inject.Instance
Caused by: java.lang.Error: java.lang.NoSuchFieldException: override
Caused by: java.lang.NoSuchFieldException: override

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,6 +1,7 @@
 language: java
 jdk:
-  - oraclejdk8
   - openjdk8
+  - openjdk11
+  - openjdk12
 script: "WELD_JUNIT_DEBUG=spotbugs mvn verify -Dspotbugs.failOnError=true"
 sudo: false

--- a/pom.xml
+++ b/pom.xml
@@ -42,13 +42,13 @@
       <!-- Versions -->
       <version.junit4>4.12</version.junit4>
       <version.junit.jupiter>5.4.1</version.junit.jupiter>
-      <version.weld>3.1.0.Final</version.weld>
+      <version.weld>3.1.1.Final</version.weld>
       <version.mockito>2.25.1</version.mockito>
       <version.jboss-ejb-api>1.0.0.Final</version.jboss-ejb-api>
       <version.hibernate-jpa-api>1.0.0.Final</version.hibernate-jpa-api>
       <!-- SpotBugs properties -->
-      <version.spotbugs>3.1.2</version.spotbugs>
-      <version.spotbugs.plugin>3.1.3</version.spotbugs.plugin>
+      <version.spotbugs>3.1.11</version.spotbugs>
+      <version.spotbugs.plugin>3.1.10</version.spotbugs.plugin>
       <spotbugs.failOnError>false</spotbugs.failOnError>
       <spotbugs.threshold>Low</spotbugs.threshold>
    </properties>

